cachito/web/api_v1.py

```diff
@@ -121,13 +121,49 @@ def create_request():
 
     # Chain tasks
     chain(
-        tasks.fetch_app_source.s(request.repo, request.ref),
-        tasks.fetch_gomod_source.s()
+        tasks.fetch_app_source.s(request.repo, request.ref).on_error(
+            tasks.set_request_state.si(request.id, 'failed')),
+        tasks.fetch_gomod_source.s().on_error(
+            tasks.set_request_state.si(request.id, 'failed')),
+        tasks.set_request_state.si(request.id, 'complete'),
     ).delay()
 
     return flask.jsonify(request.to_json()), 201
 
 
-@api_v1.errorhandler(ValidationError)
-def handle_validation_error(e):
-    return flask.jsonify(error=str(e)), 400
+@api_v1.route('/requests/<int:request_id>', methods=['PATCH'])
+def patch_request(request_id):
+    """
+    Modify the the given request.
+
+    :param int request_id: the value of the request ID
+    :return: a Flask JSON response
+    :rtype: flask.Response
+    :raise NotFound: if the request is not found
+    :raise ValidationError: if the JSON is invalid
+    """
+    # TODO: Setup authentication
+    payload = flask.request.get_json()
+    if not payload:
+        raise ValidationError('At least one key must be specified to update the request')
+
+    valid_keys = {'state'}
+    invalid_keys = set(payload.keys()) - valid_keys
+    if invalid_keys:
+        raise ValidationError(
+            'The following keys are not allowed: {}'.format(', '.join(invalid_keys)))
+
+    new_state = payload['state']
+    new_state_obj = State.query.filter_by(name=new_state).first()
+    if not new_state_obj:
+        all_state_names = sorted([state.name for state in State.query.all()])
+        raise ValidationError(
+            'The state "{}" is invalid. Please select from: {}.'
+            .format(new_state, ', '.join(all_state_names))
+        )
+
+    request = Request.query.get_or_404(request_id)
+    request.state = new_state_obj
```

Comment:
During design we talked about doing two things:
1. Add some metadata when updating the state - timestamp for now.
2. Track the history of the state. For example, when a state is updated, it's also added to `state_history`.

You could combine both items by having the `state` field simply being a property that maps to the latest state in `state_history`.

I'm fine with doing those things as a follow up PR, but I do think it's important to add this capability in the early stage of Cachito.